### PR TITLE
feat(sqlparser): support specifying DEBEZIUM JSON for CDC tables to enable schema change

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Follow up of #12164.

If we create a CDC table with..
```sql
CREATE TABLE t (..) WITH (connector = 'postgres-cdc', ..);
```
the definition will be normalized and persisted as..
```sql
CREATE TABLE t (..) WITH (connector = 'postgres-cdc', ..) FORMAT DEBEZIUM ENCODE JSON;
```
which is **not accepted** as a valid SQL because we expect `FORMAT` and `ENCODE` omitted here.

However, we'll reuse the SQL for schema change. As a result, altering a CDC table will lead to error:
```
unable to parse original table definition: 
  sql parser error:
    Row format for cdc connectors should not be set here because it is limited to debezium json
```

So this PR changes the behavior: we expect the `FORMAT` to be either omitted or exactly the same as the default value. This enables schema change on CDC tables.

----

An alternative could be introducing an `internal` flag for the parser to conditionally accept this fully qualified syntax. However, by treating omitting this clause as a syntax sugar, I guess it's more reasonable to also accept both of them.

----

Also refine the display of anyhow error, and cleanup duplicated codes.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

- Allow optional `FORMAT DEBEZIUM ENCODE JSON` after the connector definition of CDC tables.
- Allow optional `FORMAT NATIVE ENCODE NATIVE` after the connector definition of Nexmark sources or tables.

### CODE REVIEW

**Comment by BugenZhao** [comment]
Only added tests for the new syntax.

I find it painful to write e2e tests for the CDC schema change. Will investigate simplifying this procedure in the next PRs.

**Comment by tabVersion** [comment]
then how do you handle a debezium json source with schema registry 🥲

File: e2e_test/source/cdc/cdc.validate.postgres.slt [hunk 1]
```diff
@@ -125,3 +125,45 @@ create table shipments (
...
+ connector = 'postgres-cdc',
+ hostname = 'db',
+ port = '5432',
+ username = 'postgres',
+ password = 'postgres',
+ database.name = 'cdc_test',
+ schema.name = 'public',
+ table.name = 'shipments',
+ slot.name = 'shipments'
+); format debezium encode json;
```
Comment (StrikeW) [inline e2e_test/source/cdc/cdc.validate.postgres.slt:148]: The `;` before `format` seems a typo.

**Comment by BugenZhao** [comment]
> then how do you handle a debezium json source with schema registry 🥲

I'm not quite familiar with this part. Could you please elaborate more?

**Comment by BugenZhao** [comment]
> then how do you handle a debezium json source with schema registry 🥲

I guess they're unrelated? CDC sources won't leverage a schema registry.

**Review by StrikeW — APPROVED** [review]
LGTM

**Review by st1page — APPROVED** [review]

**Pull request merged**